CesiumGS/cesium
# Add geocoding credits

Cesium ion now uses Bing as the geocoder provider on the backend, and [the terms of service](https://www.microsoft.com/en-us/maps/product/enduserterms) specify that attribution must be present in any application using the service.

Unfortunately, there is no mechanism to retroactively update old versions of CesiumJS to automatically include the attribution, and if using a previous release of CesiumJS, this attribution must now be manually added to any CesiumJS app using the default Cesium ion geocoder.

This PR add attribution going forwards, but also seeks to provide a way of attributing each `GeocoderService` in a flexible manor, and to allow future changes in the default Cesium ion geocoing attribution to be reflected to clients automatically in the future

- The `Geocoder` widget will now display attributions onscreen or in the lightbox for geocoder results if present in the geocode result response under an `attributions` property.
- Otherwise a default credit from a geocoder service is shown if one is provided. This is through the property  `GeocoderService.credit` which can be implemented for each type of geocoding service accordingly.

As a part of this PR, I also moved from promise chaining to async/await where applicable to be in line with new addition to the codebase. I also cleaned up the unit tests a bit to make sure any background async operations where properly awaited.

**The Cesium ion updates are not yet live, so note that you will not see any attribution for the time being. But the code is ready for review.**

To test:
 - Get geocoding sugestions by typing in the geocoder search bar
    - A Bing attribution image should now show in the lightbox
 - Clear the suggestions by clicking 'x' or deleting all contents
    - The attribution image should be removed from the lightbox
- Type in a geocode query and hit enter or click the search icon
    - A Bing attribution image should now show in the lightbox
- Clear the searchbox by clicking 'x' or deleting all contents
   - The attribution image should be removed from the lightbox
